File: Beginner/devuTest/devuTest.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,n,ans;

	scanf("%d",&T);
	assert(T>=1 && T<=10000);

	while(T>0){

		scanf("%d",&n);
		assert(n>=1 && n<=50);

		int count[51] = {0};

		for(int i=0,d;i<n;i++){
			scanf("%d",&d);
			assert(d>=1 && d<=100);
			count[d]++;
		}

		ans = 0;

		for(int i=0;i<51;i++){
			if(count[i]>0){
				ans++;
			}
		}

		printf("%d\n",ans);

		T--;
	}

	return 0;
}

File: Beginner/devuTest/devuTest.java
import java.util.Scanner;

public class devuTest{
	public static void main(String[] args) {
		int T,n,ans;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=10000) : "T must be 1 <= T <= 10^4";

		while(T>0){

			n = scan.nextInt();
			assert(n>=1 && n<=50) : "n must be 1 <= n <= 50";

			int[] count = new int[51];

			for(int i=0;i<51;i++){
				count[i] = 0;
			}

			for(int i=0,d;i<n;i++){
				d = scan.nextInt();
				assert(d>=1 && d<=100) : "d must be 1 <= d <= 100";
				count[d]++;
			}

			ans = 0;

			for(int i=0;i<51;i++){
				if(count[i]>0){
					ans++;
				}
			}

			System.out.println(ans);

			T--;
		}


	}
}